### PR TITLE
Fix cli and logging

### PR DESCRIPTION
**Pull request by moonso**
- Refractored CLI
- Added tests for CLI
- Updated logging in CLI

### CODE REVIEW

**Comment by moonso** [comment]
@robinandeer , @guillermo-carrasco ?

File: puzzle/cli/cases.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    db_path = os.path.join(root, 'puzzle_db.sqlite3')
+    logger.info("db path is: {}".format(db_path))
+
+    if not os.path.exists(db_path):
+        logger.warn("database not initialized, run 'puzzle init'")
+        ctx.abort()
+
+    store = SqlStore(db_path)
+    for case in store.cases():
+        print(case)
```
Comment (robinandeer) [inline puzzle/cli/cases.py]: `click.echo`?

Comment (moonso) [inline puzzle/cli/cases.py]: Yes, smart

**Comment by robinandeer** [comment]
Look good to me as far as I can tell! Merge? @guillermo-carrasco?

**Comment by guillermo-carrasco** [comment]
Didn't realized you refactored the tests as well, awesome! you guys... :heart_eyes:

**Pull request merged**